Code/Framework/AzFramework/CMakeLists.txt

```diff
@@ -29,6 +29,7 @@ ly_add_target(
         PRIVATE
             AZ::AzCore
         PUBLIC
+            AZ::AtomCore.API
```

Comment:
This needs to be public because any other library that uses the TerrainDataRequestBus would end up including array_view.h. Eg: PhysX includes the TerrainDataRequestBus header from AzFramework.